### ISSUE TITLE
Make the Resource type a generic type

### ISSUE THREAD

**Issue by evert** [evert/ketting#100]

**Comment by jeffhuys** [comment]
Yes, please! We're building an application that will heavily rely on this library. It would be really nice if you'd be able to include this.

**Comment by evert** [comment]
Will do! This is low-hanging fruit. Good to know people are interested

**Comment by evert** [comment]
@jeffhuys I updated the description of this issue with the change I intend to make. Does that cover your use-case?
